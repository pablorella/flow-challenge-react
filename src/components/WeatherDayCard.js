import React from 'react';
import styled from 'styled-components';

const Card = styled.div`
  flex-basis: 25%;
  border: 1px solid #eaeaea;
  margin: 50px 30px 0 30px;
  border-radius: 10px;

  h4{
    margin-top: 0;
  }

  p {
    font-size: 0.9rem;
  }
`

const LoadingCard = styled.div`
  flex-basis: 25%;
  border: 1px solid #eaeaea;
  margin: 50px 30px 0 30px;
  border-radius: 10px;
  align-items: center;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding-bottom: 30px;

  .image {
    width: 90px;
    height: 90px;
    background: #eaeaea;
    margin: 20px;
    border-radius: 50%;
  }

  .mainTitle{
    width: 70px;
    height: 20px;
    background: #eaeaea;
  }

  .desc{
    width: 100px;
    height: 20px;
    background: #eaeaea;
    margin-top: 30px;
  }

  .data{
    width: 120px;
    height: 15px;
    background: #eaeaea;
    margin-top: 30px;
  }
`

const WeatherDay = ({weather, isFetching}) => {

  const { date, description, icon, min, max, humidity } = weather;

  if(isFetching) return <WeatherDay.Loading />
  return (
    <Card>
      <img src={`http://openweathermap.org/img/wn/${icon}@2x.png`} alt=""/>
      <h5 data-testid='date-title' >{date.getDate()} - {date.getMonth() + 1}</h5>
      <h4>{description}</h4>
      <p>Min: {min} °C</p>
      <p>Max: {max} °C</p>
      <p>Humidity: {humidity}%</p>
    </Card>
  )

}

WeatherDay.Loading = () => {
  return (
    <LoadingCard data-testid="loading-skeleton">
      <div className="image"></div>
      <div className="mainTitle"></div>
      <div className="desc"></div>
      <div className="data"></div>
      <div className="data"></div>
      <div className="data"></div>
    </LoadingCard>

  )

}

export default WeatherDay;