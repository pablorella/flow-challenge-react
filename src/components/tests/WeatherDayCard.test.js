import React from 'react';
import { render, cleanup } from '@testing-library/react';
import WeatherDayCard from '../WeatherDayCard';


describe('WeatherDayCard', () => {

  afterEach(cleanup);
  const defaultProps = {
    weather : {
      date: new Date(1584196968749 * 1000),
      description: 'some',
      icon: '10n',
      min: '2',
      max: '10',
      humidity: '30'
    },
    isFetching: false
  }

  it('should matches snapshot', () => {
    const { asFragment } = render(<WeatherDayCard {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot();
  })

  it('should render with correct content', () => {
    const { getByText, getByRole, getByTestId } = render(<WeatherDayCard {...defaultProps} />)
    const dateTitle = getByTestId('date-title');

    expect(getByRole('img')).toBeTruthy();
    expect(dateTitle).toBeTruthy();
    expect(dateTitle.textContent).toBe(`${defaultProps.weather.date.getDate()} - ${defaultProps.weather.date.getMonth() + 1}`);
    expect(getByTestId('date-title')).toBeTruthy();
    expect(getByText('some')).toBeTruthy();
    expect(getByText('Min: 2 °C')).toBeTruthy();
    expect(getByText('Max: 10 °C')).toBeTruthy();
    expect(getByText('Humidity: 30%')).toBeTruthy();
  })


  it('should render the correct content when is fetching', () => {
    const { getByTestId } = render(<WeatherDayCard {...defaultProps} isFetching={true}/>)
    expect(getByTestId('loading-skeleton')).toBeTruthy()
  })

})