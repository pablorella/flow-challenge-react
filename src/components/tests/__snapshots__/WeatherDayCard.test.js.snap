// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDayCard should matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm ibIlQz"
  >
    <img
      alt=""
      src="http://openweathermap.org/img/wn/10n@2x.png"
    />
    <h5
      data-testid="date-title"
    >
      15 - 3
    </h5>
    <h4>
      some
    </h4>
    <p>
      Min: 2 °C
    </p>
    <p>
      Max: 10 °C
    </p>
    <p>
      Humidity: 30%
    </p>
  </div>
</DocumentFragment>
`;
